File: python/calculadora/calculadora.py
class Calculadora:
    def adicionar(self, a, b):
        return a + b

    def subtrair(self, a, b):
        return a - b

    def multiplicar(self, a, b):
        return a * b

    def dividir(self, a, b):
        if b == 0:
            raise ValueError("Divisão por zero não é permitida")
        return a / b

    def potencia(self, base, expoente):
        return base ** expoente

    def raiz_quadrada(self, numero):
        if numero < 0:
            raise ValueError("Raiz quadrada de número negativo não é permitida")
        return numero ** 0.5

File: javascript/calculadora.js
class Calculadora {
    adicionar(a, b) {
        return a + b;
    }

    subtrair(a, b) {
        return a - b;
    }

    multiplicar(a, b) {
        return a * b;
    }

    dividir(a, b) {
        if (b === 0) {
            throw new Error("Divisão por zero não é permitida");
        }
        return a / b;
    }

    potencia(base, expoente) {
        return Math.pow(base, expoente);
    }

    raizQuadrada(numero) {
        if (numero < 0) {
            throw new Error("Raiz quadrada de número negativo não é permitida");
        }
        return Math.sqrt(numero);
    }
}

module.exports = Calculadora;
